src/external/token/NativeIssuance_v1.sol

```diff
@@ -0,0 +1,238 @@
+ *               - The `depositNative` function is used to deposit native tokens into the contract, which can then be burned by calling the `burn` function to send them to the burn address.
+ * @author Inverter Network
+ */
+contract NativeIssuance_v1 is INativeIssuance_v1, IERC20Metadata, Ownable {
+    using Address for address payable;
+
+    /**
+     * @notice The address of the native minter precompile.
+     */
+    INativeMinter public constant NATIVE_MINTER =
+        INativeMinter(0x0200000000000000000000000000000000000001);
+
+    /**
+     * @notice The address where native tokens are sent in order to be burned for the bonding curve.
+     *
+     * @dev This address was chosen arbitrarily.
+     */
+    address public constant BURNED_FOR_BONDINGCURVE_ADDRESS =
+        0x0100000000000000000000000000000000010203;
+
+    // State Variables
+
+    /**
+     * @notice Total number of tokens minted by this contract through the native minter precompile.
+     */
+    uint public totalMinted;
+
+    /**
+     * @notice Mapping of the amount of native tokens deposited for burning by an address.
+     */
+    mapping(address => uint) public depositsForBurning;
+
+    /// @dev    The mapping of allowed minters.
+    mapping(address => bool) public allowedMinters;
+
+    //------------------------------------------------------------------------------
+    // Modifiers
+
+    /// @dev    Modifier to guarantee the caller is a minter.
+    modifier onlyMinter() {
+        if (!allowedMinters[_msgSender()]) {
+            revert IERC20Issuance__CallerIsNotMinter();
+        }
+        _;
+    }
+
+    //------------------------------------------------------------------------------
+    // Constructor
+    constructor(address initialOwner) Ownable(initialOwner) {
+        _setMinter(initialOwner, true);
+    }
+
+    receive() external payable {}
+    fallback() external payable {}
+
+    /**
+     * @notice Mints native tokens to the specified address.
+     * @param  to   The address that will receive the minted tokens.
+     * @param  amount   The number of tokens to be minted.
+     */
+    function mint(address to, uint amount) external override onlyMinter {
+        if (to == address(0)) {
+            revert INativeIssuance_v1__InvalidAddress();
+        }
+        // require(amount > 0, "Amount must be greater than zero");
+
+        emit Minted(to, amount);
+
+        totalMinted += amount;
+
+        // Calls NativeMinter precompile through INativeMinter interface.
+        NATIVE_MINTER.mintNativeCoin(to, amount);
+    }
+
+    /**
+     * @notice Deposits native tokens to be able to burn native tokens from the specified address.
+     * @dev    `msg.value` is the amount of native tokens to be deposited.
+     * @param  from   The address that will have native tokens deposited for.
+     */
+    function depositNative(address from) external payable {
```

Comment:
Out of curiosity: Why do this as a separate function call in the bonding curve override, instead of calling it from inside the burn() function in this contract?

Reply:
This is a payable function because I needed to burn native tokens. I implemented it to comply with the current interfaces: `IERC20Issuance_v1`, `IRedeemingBondingCurveBase_v1`, and `IBondingCurveBase_v1`.

To burn tokens, the user needs to transfer native tokens directly to the contract. The current implementation works with ERC20 tokens and uses the approve/allowance mechanism to transfer tokens after user approval. 

In the funding manager, I added a payable `sellNative` function, where users send native tokens to be burned. This function calls `depositNative`, which sends the tokens to `NativeIssuance`, responsible for burning them.

I’m not entirely happy with this approach, but it allowed me to implement it without breaking the existing interfaces or rewriting the burning mechanism from scratch.

Reply:
I find the whole topic of adapting our system, which was designed in a way that it only takes in ERC20 tokens, to take native tokens kinda difficult to access. @DogaOztuzun did you have the chance to talk about this to anybody else from the Contract Team?